### DISCUSSION TITLE
Passing a string to

### DISCUSSION THREAD

**Discussion by fredb-gc** [pydantic/pydantic#4425] [Question] [answered]
Hello, I've had multiple attempts to try and figure this out but I haven't got anywhere. I'm using Pydantic for settings management, and currently I pass a `Dict()` object to my functions which have the `@validate_arguments` decorator. Each of my `Dict()` objects are actually JSON files that I read via:
```python
from json import load

with open("my-config.json") as f:
    config = load(f)

```

To make it easier for people to use my code, I would like to skip a step and allow users to do pass a file path as their configuration instead of a `Dict()` object e.g. I would like to change this:

```python
from json import load
from pydantic import BaseModel, validate_arguments

class Config(BaseModel):
    foo: str
    bar: int

@validate_arguments
def my_func(config: Config):
    pass

with open("my-config.json") as f:
    config = load(f)

my_func(config)

```

To this:

```python
from pydantic import BaseModel, validate_arguments

class Config(BaseModel):
    foo: str
    bar: int

@validate_arguments
def my_func(config: Config):
    pass

my_func("my-config.json") # Pass the filename here instead of the Dict() object
```

The `.env` support is very close to what I'm looking for and the [`parse_file`](https://pydantic-docs.helpmanual.io/usage/models/#helper-functions) helper function has all the functionality I need. But I would have to update my code to be:

```python
from pydantic import BaseModel, validate_arguments

class Config(BaseModel):
    foo: str
    bar: int

@validate_arguments
def my_func(config: Config):
    pass

my_func(Config.parse_file("my-config.json")) # Not the most succinct and friendly solution
```

Has anyone done something similar and has a solution? Thanks!

**Comment by samuelcolvin** [comment, marked as answer]
This should be possible with your own function decorator, but it's outside the bounds of what pydantic itself would do.

**Reply by fredb-gc** [reply]
Thank you! Wasn't sure that I was missing some hidden functionality
